components/brave_sync/time_limited_words_unittest.cc

```diff
@@ -0,0 +1,185 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
```

Comment:
This is a really thorough and excellent test suite. It makes the review so much easier, thanks!

Reply:
thank you for the good words :)
and the tests helped me several times to find that my changes would break some border cases